### PR TITLE
WebHost: unify references to options

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
Unify references to options in the webhost as options, instead of using options and settings interchangeably.

## How was this tested?
Running webhost and viewing the changes. Weighted options page is currently broken so putting in draft for now.

## If this makes graphical changes, please attach screenshots.
Screenshot_49
Screenshot_51
Screenshot_52

### CODE REVIEW

**Comment by MrRedstone54** [comment]
On the third image it shows "download a options file." This should be, "download an options file". Since this draft is about language, might as well make sure the grammar is correct.

**Comment by alwaysintreble** [comment]
@LegendaryLinux

**Comment by LegendaryLinux** [comment]
Semantically there is a difference between a setting and an option. A setting has both a name and a set of options.

I also don't want to change the title of the pages away from "Player Settings" and "Weighted Settings."

In your first image, the only inconsistency is the header "Game Options", which should be changed to "Game Settings." The rest is correct as written.

**Review by LegendaryLinux — CHANGES_REQUESTED** [review]
See my above comment.

**Comment by alwaysintreble** [comment]
> Semantically there is a difference between a setting and an option. A setting has both a name and a set of options.
> 
> I also don't want to change the title of the pages away from "Player Settings" and "Weighted Settings."
> 
> In your first image, the only inconsistency is the header "Game Options", which should be changed to "Game Settings." The rest is correct as written.

So, semantically, I agree. I should've more clearly outlined the goal/reasoning in the original post so that's on me. Previously, in the API we've been kind of using settings and options interchangeably, but as of https://github.com/ArchipelagoMW/Archipelago/commit/827444f5a4e065bc310c889892580eb1f97c73cb this is no longer the case. We now have a clear cut line in the actual Archipelago API on wh at an "option" and a "setting" are, used to define 2 separate interactions, where an option is a player submitted choice for their world, and a setting is the host configured choices for the multiworld, and the local settings used to configure their Archipelago installation (ROM paths etc). Primary goal of this PR is to unify these terms so that any references to options are called options, and references to settings are settings. I'm open to compromise if you still believe it, but context is important here.

File: WebHostLib/misc.py [hunk 1]
```diff
@@ -36,15 +36,15 @@ def start_playing():
...
 
 
-# Player settings pages
-@app.route('/games/<string:game>/player-settings')
-def player_settings(game):
-    return render_template(f"player-settings.html", game=game, theme=get_world_theme(game))
+# Player options pages
+@app.route('/games/<string:game>/player-options')
+def player_options(game):
+    return render_template(f"player-options.html", game=game, theme=get_world_theme(game))
```
Comment (black-sliver) [inline WebHostLib/misc.py:47]: I would like to see a flask.redirect from the old url to the new one with code 301 Moved Permanently for a version or two, so link in browser history get updated. (I have not checked if the site map "breaks" doing that)

**Review by black-sliver — COMMENTED** [review]
The change to use consistent language is very welcome from a my point of view. When people talk about "player options" it should be clear that the game's option api will be used. When people talk about "game settings" it should be clear that we talk about the game's settings api/host.yaml.

"Player settings" is unclear in that regard.

**Review by LegendaryLinux — APPROVED** [review]
Finally found time to go through this. Looks okay to me.

File: WebHostLib/misc.py [hunk 2]
```diff
@@ -37,17 +37,31 @@ def start_playing():
     return render_template(f"startPlaying.html")
 
 
+# TODO for back compat. remove around 0.4.5
 @app.route('/weighted-settings')
 @cache.cached()
-def weighted_settings():
-    return render_template(f"weighted-settings.html")
+def route_weighted_options():
+    return redirect("/weighted-options", 301)
```
Comment (black-sliver) [inline WebHostLib/misc.py:44]: I think this is how it's supposed to be done:
```suggestion
    return redirect(url_for('player_options', game=game), 301)
```

**Review by black-sliver — CHANGES_REQUESTED** [review]

**Comment by black-sliver** [comment]
Hm, we use `"` in AP, not `'`. Can you change the changed strings? I realize that my suggestion was wrong - have been working on a different code base at the time :S

**Comment by alwaysintreble** [comment]
I did `'` not because of your suggestion, but because *most* of that file uses `'`. Not all of it does though so I suppose this could go either way.

**Comment by black-sliver** [comment]
I think the decision we reached a while ago was to change all touched strings from `'` to `"`, but not change strings that are not touched.

File: WebHostLib/misc.py [hunk 3]
```diff
@@ -37,17 +37,31 @@ def start_playing():
     return render_template(f"startPlaying.html")
 
 
-@app.route('/weighted-settings')
+# TODO for back compat. remove around 0.4.5
+@app.route("/weighted-settings")
 @cache.cached()
-def weighted_settings():
-    return render_template(f"weighted-settings.html")
+def route_weighted_options():
```
Comment (black-sliver) [inline WebHostLib/misc.py:43]:
```suggestion
def weighted_settings():
```
also `@cache.cached()` should go, sadly i can't make a suggestion for this

File: WebHostLib/misc.py [hunk 4]
```diff
@@ -37,17 +37,31 @@ def start_playing():
...
-def player_settings(game):
-    return render_template(f"player-settings.html", game=game, theme=get_world_theme(game))
+def weighted_options():
+    return render_template("weighted-options.html")
+
+
+# TODO for back compat. remove around 0.4.5
+@app.route("/games/<string:game>/player-settings")
+@cache.cached()
+def route_player_options(game: str):
```
Comment (black-sliver) [inline WebHostLib/misc.py:56]:
```suggestion
@app.route("/games/<string:game>/player-settings")
def player_settings(game: str):
```

**Review by black-sliver — CHANGES_REQUESTED** [review]

**Review by black-sliver — APPROVED** [review]
Have not checked js and css changes, only py and checked the links and redirects

**Pull request merged**